ivy/functional/backends/tensorflow/experimental/statistical.py

```diff
@@ -160,6 +161,125 @@ def corrcoef(
+    dtype = input.dtype
+    temp = tf.cast(input, tf.float64)
+    num_dim = tf.rank(temp)
+    keepdim_shape = tf.shape(temp)
+    q = 50.0
+
+    if axis is None:
+        if tf.reduce_all(tf.math.is_nan(temp)).numpy():
+            temp = tf.reshape(temp, shape=(1, -1))
+        else:
+            temp = temp[~tf.math.is_nan(temp)]
+
+        ret = tfp.stats.percentile(
+            temp,
+            q,
+            axis=axis,
+            interpolation="midpoint",
+            keepdims=keepdims,
+        )
+        ret = tf.cast(ret, dtype=dtype)
+        return ret
+
+    if isinstance(axis, tuple):
+        if len(axis) == 1:
+            axis = axis[0]
+
+    if isinstance(axis, int):
+        axis = [axis]
+    if isinstance(axis, tuple):
+        axis = list(axis)
+
+    for i in axis:
+        keepdim_shape = tf.tensor_scatter_nd_update(keepdim_shape, [[i]], [1])
+    axis = [num_dim + x if x < 0 else x for x in axis]
+    axis.sort()
+    dimension = tf.size(temp.shape)
+    while tf.size(axis) > 0:
+        axis1 = axis[0]
+        for axis2 in range(axis1 + 1, dimension):
+            temp = tf.transpose(
+                temp,
+                perm=tf.tensor_scatter_nd_update(
+                    tf.range(tf.rank(temp)), [[axis1], [axis2]], [axis2, axis1]
+                ),
+            )
+            axis1 = axis2
+        axis = [x - 1 for x in axis]
+        axis.pop(0)
+        dimension = dimension - 1
+    temp = tf.reshape(
+        temp, shape=tf.concat([tf.shape(temp)[: (dimension - len(axis))], [-1]], axis=0)
+    )
+    temp = tf.cast(temp, dtype)
+
+    tensor = tf.reshape(temp, shape=(1, -1))
+    shape = temp.shape
+    dim = temp.ndim
+    slice_size = shape[len(shape) - 1]
+    num_jumps = 1
+    result = []
+
+    if slice_size == 1:
+        if dim == 2 and input.shape[0] == 1:
+            return tensor
+        if dim > 2 and input.shape[0] == 1:
+            return tf.reshape(tensor, shape=input.shape)
+
+        tensor = tf.reshape(tensor, shape=shape[:-1])
+        return tensor
+
+    i = dim
+    while i > 1:
+        num_jumps *= shape[len(shape) - i]
+        i -= 1
+
+    for i in range(num_jumps):
+        start = i * slice_size
+        end = (i + 1) * slice_size
+        arr = tensor[:, start:end]
+        if tf.reduce_all(tf.math.is_nan(arr)).numpy():
+            arr = tf.reshape(arr, shape=(1, -1))
+        else:
+            arr = arr[~tf.math.is_nan(arr)]
+
+        ret = tfp.stats.percentile(
+            arr, q, axis=None, interpolation="midpoint", keepdims=keepdims
+        ).numpy()
+        if keepdims:
+            ret = set_value(ret)
+
+        result.append(ret)
+
+    result = tf.reshape(result, shape=shape[:-1]).numpy()
+
+    if keepdims:
+        keepdim_shape = tuple(keepdim_shape)
+        result = tf.reshape(result, shape=keepdim_shape)
+        result = tf.cast(result, dtype)
+
+    if input.dtype in [tf.int64, tf.float64]:
```

Comment:
any particular reason to have these `dtype-specific` checks and type castings? Also, why not return a `tf.Tensor` using something like `tf.constant(result, dtype= input.dtype)` for example? I think `tf.constant` is generally more stable across different versions of TF and so it's better to avoid resorting to an experimental module unless there's a specific reason to do so 🙂

Reply:
the specific `type-casting` was to make it compatible with the `numpy` because `numpy` somehow returns the fixed types on multiple input types, so to make it compatible with `numpy`, I used this. I added these when testing multiple edge cases, so maybe something slipped from my mind, I'll test it and change it if it needs to be changed and yup, I didn't think about using `tf.constant`, I'll use this after testing it and I hope it should work fine 👍.

Reply:
Just another point regarding this, I don't think there is a need to convert the `tf.Tensors` to `np.ndarrays` throughout the code. Doing these `.numpy()` calls only adds to the computational overhead so its better to stick to native tensors 🙂

Reply:
I removed the `.numpy()` calls all over the function, and now it seems fine. I also updated the `dtype-casting` using `tf.cast` which looks more familiar in this scenario. About using checks there are reasons which are `numpy` always returns `float64` when we have these dtypes : `int32`, `int64`, and `float64`. if we have `float32` it returns the same.